hazelcast/src/test/java/com/hazelcast/map/LocalMapStatsTest.java

```diff
@@ -29,6 +34,25 @@
 public class LocalMapStatsTest extends HazelcastTestSupport {
 
     @Test
+    public void testMapStatsAreNotLeaking() throws Exception {
```

Comment:
this test as it is is failing on the current master -> it shows the stats. data are retained even when the map is destroyed -> leak. 

my approach is not bullet-proof and I understand your concert about a hidden leak. yours is more fragile as it makes more assumptions about implementation. It's hard to tell what's better -> :+1: from me and the decision what approach is better on you.